patches/net/minecraft/server/level/ChunkMap.java.patch

```diff
@@ -97,6 +97,27 @@
              EntityType<?> entitytype = p_140200_.getType();
              int i = entitytype.clientTrackingRange() * 16;
              if (i != 0) {
+@@ -1260,6 +_,20 @@
+         });
+     }
+ 
++    // Neo: Getter for players watching an entity.
++    public List<ServerPlayer> getPlayersWatching(Entity entity) {
++        var trackedEntity = entityMap.get(entity.getId());
++        if (trackedEntity != null) {
++            var ret = new java.util.ArrayList<ServerPlayer>(trackedEntity.seenBy.size());
++            for (var connection : trackedEntity.seenBy) {
++                ret.add(connection.getPlayer());
++            }
++            return List.copyOf(ret);
```

Comment:
Why bother returning a copy here? This is a brand-new list

Reply:
I wanted to keep it immutable from the outside. Probably overkill though.

Reply:
Return an unmodifiable view?

Reply:
OK made it an unmodifiable list.